src/create-pr.js

```diff
@@ -0,0 +1,271 @@
+const FILE_UPDATE_COMMIT_DESC = core.getInput('FILE_UPDATE_COMMIT_DESC');
+const RELEASE_PR_IDENTIFIER_LABEL = core.getInput('RELEASE_PR_IDENTIFIER_LABEL');
+const RELEASE_PR_TITLE = core.getInput('RELEASE_PR_TITLE');
+const COMMIT_USERNAME = 'github-actions[bot]';
+const COMMIT_USEREMAIL = '<41898282+github-actions[bot]@users.noreply.github.com>';
+
+async function run () {
+  try {
+    owner = getOwner();
+    repo = getRepoName();
+
+    await cloneRepo();
+
+    const lastRelease = await getLastRelease();
+    const filteredPullRequests = await getPullRequests(lastRelease)
+
+    if (filteredPullRequests.length === 0 ) {
+      throw new Error('No Pull requests has been merged since last release');
+    }
+
+    await createReleaseCandidateBranch();
+    const pr = await createReleasePR(getReleasePRBody(filteredPullRequests));
+    await addLabelToThePR(pr.number);
+
+    console.debug(`Pull Request Created: ${pr.html_url}`);
+  } catch (error) {
+    core.setFailed(error.message);
+  }
+}
+
+/**
+ * Add Release Label to the RC PR
+ *
+ * @param {string} prNumber pull request number
+ */
+async function addLabelToThePR (prNumber) {
+  await octokit.issues.update({
+    owner,
+    repo,
+    issue_number: prNumber,
+    labels: [RELEASE_PR_IDENTIFIER_LABEL]
+  });
+}
+
+/**
+ * Clone the repository
+ */
+async function cloneRepo () {
+  const REPO_URL = 'https://' + owner + ':' + GITHUB_TOKEN + '@github.com/' + owner + '/' + repo;
+
+  await simpleGit(GITHUB_WORKSPACE)
+    .clone(REPO_URL);
+}
+
+/**
+ * Create Release Candidate Branch with a new Commit
+ */
+async function createReleaseCandidateBranch () {
+  const cwd = GITHUB_WORKSPACE + '/' + repo;
+
+  _.each(REPLACE_COMMANDS, function (cmd) {
+    cmd = cmd.replace('<<DATE>>', dayjs().format('YYYY-MM-DD'));
+    execSync(cmd, { cwd });
+  });
+
+  await simpleGit(cwd)
+    .addConfig('user.name', COMMIT_USERNAME)
+    .addConfig('user.email', COMMIT_USEREMAIL)
+    .checkoutLocalBranch(getRcBranchName())
+    .add('./*')
+    .commit(FILE_UPDATE_COMMIT_DESC)
+    .push('origin', getRcBranchName())
+}
+
+/**
+ * Create Release Pull Request
+ *
+ * @param {string} body body of the pr to be created
+ * @returns {Promise} newly created pr object
+ */
+async function createReleasePR (body) {
+  var pr = await octokit.pulls.create({
+    owner,
+    repo,
+    title: RELEASE_PR_TITLE,
+    head: getRcBranchName(),
+    base: BASE_BRANCH,
+    body
+  });
+
+  return pr.data;
+}
+/**
+ * Get Release Candidate Branch Name
+ *
+ * @returns {string} branch name
+ */
+function getRcBranchName () {
+  return RELEASE_VERSION + '-rc';
+}
+
+/**
+ * Get Release Pull Request body content
+ *
+ * @param {Array} mergedPRs list of merged prs
+ * @returns {string} body content
+ */
+function getReleasePRBody (mergedPRs) {
+  const date = dayjs().format('DD MMMM, YYYY');
+  var body = `## Release Update - ${date}\n\n### Changelog\n`;
+
+  _.each(mergedPRs, function (pr) {
```

Comment:
arrays already have a `.forEach` native method.

You can also do something like this:

```js
const date = dayjs().format('DD MMMM, YYYY');
const releaseTitle = `## Release Update - ${date}\n\n### Changelog\n`;
const prTitles = mergedPRs.map((pr) =>  `\n* ${pr.title} #${pr.number} - @${pr.user}`;

return releaseTitle + prTitles.join('');
```